src/world/FieldComputations.cpp

```diff
@@ -185,125 +94,113 @@ std::shared_ptr<Vector2> FieldComputations::lineIntersectionWithDefenceArea(cons
 
     if (intersections.size() == 1) {
         return std::make_shared<Vector2>(intersections.at(0));
-    } else if (intersections.size() > 1) {
-        double closestIntersectionToLineStart = INT_MAX;
-        Vector2 closestIntersection = intersections.at(0);
-        for (auto const &intersection : intersections) {
-            if (lineStart.dist(intersection) < closestIntersectionToLineStart) {
-                closestIntersection = intersection;
-                closestIntersectionToLineStart = lineStart.dist(intersection);
-            }
-        }
-        return std::make_shared<Vector2>(closestIntersection);
+    } else if (intersections.size() == 2) {
+        double distanceFirstIntersection = lineStart.dist(intersections.at(0));
+        double distanceSecondIntersection = lineStart.dist(intersections.at(1));
+        return std::make_shared<Vector2>(distanceFirstIntersection < distanceSecondIntersection ? intersections.at(0) : intersections.at(1));
```

Comment:
no, no no no no no no nono.

std::optional<Vector2>...

Also, operator[]

Reply:
In this case I stick to the old choice, because it cost me more time to refactor it to std::optional. Also programmers often use nullptr as some kind of optional variable, so I currently don't see why it is that bad. But if you show me your arguments why nullptr is worse than an std::optional then I probably quickly agree with your reasoning.